src/Converter_TEST.cc

```diff
@@ -2141,6 +2141,281 @@ TEST(Converter, Pose_16_to_17)
+  convertedElem = convertedElem->FirstChildElement();
+  EXPECT_STREQ(convertedElem->Name(), "pose");
+  EXPECT_STREQ(convertedElem->Attribute("relative_to"), "__model__");
+
+  convertedElem = convertedElem->NextSiblingElement();
+  EXPECT_STREQ(convertedElem->Name(), "model");
+  EXPECT_STREQ(convertedElem->Attribute("canonical_link"), "C");
+  EXPECT_EQ(convertedElem->NextSiblingElement(), nullptr);
+  convertedElem = convertedElem->FirstChildElement();
+  EXPECT_STREQ(convertedElem->Name(), "pose");
+  EXPECT_STREQ(convertedElem->Attribute("relative_to"), "__model__");
+  convertedElem = convertedElem->NextSiblingElement();
+  EXPECT_STREQ(convertedElem->Name(), "link");
+  EXPECT_STREQ(convertedElem->Attribute("name"), "C");
+  EXPECT_EQ(convertedElem->NextSiblingElement(), nullptr);
+  convertedElem = convertedElem->FirstChildElement();
+  EXPECT_STREQ(convertedElem->Name(), "pose");
+  EXPECT_STREQ(convertedElem->Attribute("relative_to"), "__model__");
+
+  // Another flattened world in 1.7 format
+  xmlString = R"(
+  <?xml version="1.0" ?>
+  <sdf version='1.7'>
+    <world name="default">
+      <model name="ParentModel">
+        <frame name="ChildModel::__model__" attached_to="ChildModel::L1">
+          <pose relative_to="__model__">1 0 1 0 0 0</pose>
+        </frame>
+        <frame name="ChildModel::NewFrame" attached_to="ChildModel::L1">
+          <pose relative_to="ChildModel::Something">1 0 1 0 0 0</pose>
+        </frame>
+        <link name="ChildModel::L1">
+          <pose relative_to="ChildModel::__model__">0 1 0 0 0 0</pose>
+          <visual name="v1">
+            <geometry>
+              <sphere>
+                <radius>0.1</radius>
+              </sphere>
+            </geometry>
+          </visual>
+        </link>
+        <link name="ChildModel::L2">
+          <pose relative_to="ChildModel::__model__">0 0 0 0 0 0</pose>
+        </link>
+        <joint name="ChildModel::J1" type="revolute">
+          <parent>ChildModel::L1</parent>
+          <child>ChildModel::L2</child>
+        </joint>
+        <joint name="ChildModel::J2" type="revolute">
+          <pose relative_to="ChildModel::__model__">0 0 0 0 0 0</pose>
+          <parent>ChildModel::L1</parent>
+          <child>ChildModel::L2</child>
+          <axis>
+            <xyz expressed_in="ChildModel::NewFrame">0 0 1</xyz>
+          </axis>
+          <axis2>
+            <xyz expressed_in="ChildModel::NewFrame">0 0 1</xyz>
+          </axis2>
+          <sensor name="camera" type="camera">
+            <pose relative_to="ChildModel::NewFrame">1 0 0 0 0 0</pose>
+          </sensor>
+        </joint>
+        <gripper name="gripper">
+          <gripper_link>ChildModel::L1</gripper_link>
+          <palm_link>ChildModel::L2</palm_link>
+        </gripper>
+      </model>
+    </world>
+  </sdf>)";
+
+  xmlDoc.Clear();
+  xmlDoc.Parse(xmlString.c_str());
+
+  sdf::Converter::Convert(&xmlDoc, &convertXmlDoc);
```

Comment:
For readability, could you also add a mutliline test on generated XML?

While the below `EXPECT*` statements reallllly drill into semantics, it makes it a tad harder to quickly assess that the spec example is fully represented w/ this implementation.

Reply:
9ed27130e8343867e79d88dae4be3894a7f0ac82